.github/workflows/sv_pipeline_docker.yml

```
@@ -43,6 +50,92 @@ jobs:
           python -m pip install --upgrade pip
           pip install termcolor
 
+      - name: Determine Target Images
+        id: target_images
+        # This step determines the target images to be rebuilt based
+        # on the files changed between the HEAD and BASE commits, where
+        # HEAD is the commit that its push triggered running this action and
+        # BASE is the commit the PR is targeting. For instance, if the
+        # commit changes the `dockerfiles/sv-base/Dockerfile`, then the
+        # `sv-base` docker image (and all its dependencies) are rebuilt.
+        #
+        # This step first gets the HEAD and BASE commit SHAs from the
+        # $GITHUB_CONTEXT environment variable. Then uses `git diff` to
+        # determine the files changed between the commits. Based on the
+        # changed files, it determines a list of docker images to be rebuilt.
+        # Finally, it stores the determined docker images in $TARGETS
+        # variable that can be accessed in other steps.
+        run: |
+          # The commit SHA of the base branch which this PR targets;
+          # ideally broadinstitute/gatk-sv:master.
+          BASE_SHA=$(echo "$GITHUB_CONTEXT"| jq -r '.event.pull_request.base.sha')
+
+          # The commit SHA of the commit that triggered the action.
+          HEAD_SHA=${{ github.event.pull_request.head.sha }}
+
+          # A list of all the files changed in HEAD commit w.r.t BASE commit.
+          CHANGED_FILES=$(git diff --name-only $BASE_SHA $HEAD_SHA)
+
+          # Add a given target image name to the TARGETS
+          # array if it is not already in the array.
+          TARGETS=()
+          try_add_target() {
+            if [[ ! " ${TARGETS[*]} " =~ $1 ]]; then
+              TARGETS+=($1)
+            fi
+          }
+
+          for i in "${CHANGED_FILES[@]}"
+          do
+            if [[ $i == *".github/workflows/sv_pipeline_docker.yml"* ]]; then
+              # A change to this file may impact how images are built,
+              # hence, if this file is changed, all the images are rebuilt.
+              try_add_target "all"
```

Right now build_docker.py throws an error if "all" is in the targets list any anything else is too, which seems like it could happen depending on the order of changed files.. So this should be changed to
```suggestion
              TARGETS=("all")
```

---

Good point, thank you.